File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;
        int j = nums.size()-1; 
        while (i < j) { 
            if (nums[i] + nums[j] > target) { 
                j--;
                continue;
            }
            if (nums[i] + nums[j] < target) { 
                i++; 
                continue;
            }
            return vector<int> {i+1, j+1};
        }
        return vector<int> {}; 
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # target = nums[i] + nums[j], diff = nums[j] = target - nums[i]
        differences_map = {}
        for i in range(0, len(nums)):
            if nums[i] in differences_map:
                return [differences_map[nums[i]], i]
            
            difference = target - nums[i] #current number difference
            differences_map[difference] = i